server/bin/index-pbench

```diff
@@ -2095,112 +2550,299 @@ class PbenchTarBall(object):
+        { "directory": "/rr/td/host.example.com/",                "mode": 0o755, "mtime": "2018-10-24T10:58:04" }
+        { "directory": "/rr/td/host.exmaple.com/iostat/",         "mode": 0o755, "mtime": "2018-10-24T10:58:05",
+            "files": [
+                { "name": "iostat.cmd",             "size":   48, "mode": 0o755, "mtime": "2018-10-24T10:38:06" },
+                { "name": "iostat-stdout.txt",      "size": 7598, "mode": 0o644, "mtime": "2018-10-24T10:58:07" },
+                { "name": "disk-average.txt",       "size":  382, "mode": 0o644, "mtime": "2018-10-24T10:58:11" },
+                { "name": "disk.html",              "size": 1379, "mode": 0o644, "mtime": "2018-10-24T10:58:12" }
+            ] }
+        { "directory": "/rr/td/host.exmaple.com/iostat/csv/",     "mode": 0o755, "mtime": "2018-10-24T10:58:08",
+            "files": [
+                { "name": "disk_IOPS.csv",           "size": 512, "mode": 0o644, "mtime": "2018-10-24T10:58:09" },
+                { "name": "disk_Wait_Time_msec.csv", "size": 512, "mode": 0o644, "mtime": "2018-10-24T10:58:10" }
+            ] }
         """
-        # Fetch the original values from the metadata.log file.
-        start_run_orig = self.mdconf.get('run', 'start_run')
-        end_run_orig = self.mdconf.get('run', 'end_run')
-        date_orig = self.mdconf.get('pbench', 'date')
-        # We'll accept dates that match any of the following:
-        #  * 2019-01-10_12:12:12
-        #  * 2019-01-10T12:12:12
-        #  * 20190110_12:12:12
-        #  * 20190110T12:12:12
-        #  * 2019-01-10_12:12:12.123456[789]
-        #  * 2019-01-10T12:12:12.123456[789]
-        #  * 20190110_12:12:12.123456[789]
-        #  * 20190110T12:12:12.123456[789]
-        # However, we reduce the combinations by immediately replacing the
-        # "_" version with a "T", and ensure we only keep microseconds of
-        # resolution, dropping any extra nanoseconds because that level of
-        # precision is ignored and causes problems during date/time string
-        # conversions (e.g. Elasticsearch ignores it, and python std lib
-        # does not offer a way to handle nanoseconds).
-        formats = [ "%Y-%m-%dT%H:%M:%S.%f", "%Y%m%dT%H:%M:%S.%f", "%Y-%m-%dT%H:%M:%S", "%Y%m%dT%H:%M:%S" ]
-        def convert_to_dt(dt_str):
-            rts = dt_str.replace('_', 'T')
-            us_offset = rts.rfind('.')
-            us = rts[us_offset:][:7]
-            rts = rts[:us_offset] + us
-            for f in formats:
+        prefix_l = len(self.dirname)
+        # Note we have to fully populate the dictionary by processing all the
+        # members before we can yield the generated sources for each
+        # directory.
+        toc_dirs = _dict_const()
+        for m in self.members:
+            # Always strip the prefix
+            name = m.name[prefix_l:]
+            if m.isdir():
+                dname = name if name.endswith(os.path.sep) else "{}/".format(name)
```

Comment:
For consistency, use '/' in both places or os.path.sep in both.

Reply:
Fixed.